app/assets/javascripts/member-facing/redirectors.js

```diff
@@ -0,0 +1,43 @@
+const RegisterMemberRedirector = {
+  attemptRedirect:  function(followUpUrl, member) {
+
+    if(typeof(member) !== 'object') {
+      member = window.champaign.personalization.member;
+    }
+
+    if ( shouldMemberRegister() ) {
+      redirectToRegistration();
+      return true;
+    } else {
+      return false;
+    }
+
+    function shouldMemberRegister() {
+      return !member.registered;
+    }
+
+    function redirectToRegistration() {
+      var url = `/member_authentication/new?follow_up_url=${encodeURIComponent(followUpUrl)}&email=${encodeURIComponent(member.email)}`;
+      redirectTo(url);
+    }
+  }
+};
+
+const AfterDonationRedirector = {
+  attemptRedirect: function (followUpUrl, donationFormData) {
+    if(!(donationFormData.storeInVault && RegisterMemberRedirector.attemptRedirect(followUpUrl, donationFormData.member))) {
```

Comment:
this is a little hard to parse. are you using the `&&` to ensure that it only attempts to call the Redirector if storeInVault is true? I think this would be less succinct but much clearer as
```
if (donationFormData.storeInVault) {
  redirected = RegisterMemberRedirector.attemptRedirect(followUpUrl, donationFormData.member);
  if (!redirected) redirectTo(followUpUrl);
else {
  redirectTo(followUpUrl);
}
```